packages/ui-table/src/Table/README.md

```diff
@@ -167,141 +351,412 @@ class Example extends React.Component {
+        {
+          id: 'year',
+          text: 'Year',
+          width: '15%'
+        },
+        {
+          id: 'rating',
+          text: 'Rating',
+          width: '15%',
+          renderCell: (rating) => rating.toFixed(1)
+        }
+      ]}
+      rows={[
+        {
+          id: '1',
+          rank: 1,
+          title: 'The Shawshank Redemption',
+          year: 1994,
+          rating: 9.3
+        },
+        {
+          id: '2',
+          rank: 2,
+          title: 'The Godfather',
+          year: 1972,
+          rating: 9.2
+        },
+        {
+          id: '3',
+          rank: 3,
+          title: 'The Godfather: Part II',
+          year: 1974,
+          rating: 9.0
+        },
+        {
+          id: '4',
+          rank: 4,
+          title: 'The Dark Knight',
+          year: 2008,
+          rating: 9.0
+        },
+        {
+          id: '5',
+          rank: 5,
+          title: '12 Angry Men',
+          year: 1957,
+          rating: 8.9
+        }
+      ]}
+    />
+  )
+  ```
 
+- ```javascript
+  const SortableTable = ({ caption, headers, rows }) => {
     const initialColWidth = {}
     headers.forEach((header) => {
       initialColWidth[header.id] = 'start'
     })
 
-    this.state = {
-      sortBy: headers && headers[0] && headers[0].id,
-      ascending: true,
-      colTextAligns: initialColWidth
-    }
-  }
+    const [sortBy, setSortBy] = useState(headers && headers[0] && headers[0].id)
+    const [ascending, setAscending] = useState(true)
+    const [colTextAligns, setColTextAligns] = useState(initialColWidth)
 
-  handleSort = (event, { id }) => {
-    const { sortBy, ascending } = this.state
+    const sortedRows = useMemo(() => {
+      if (!sortBy) return rows
 
-    if (id === sortBy) {
-      this.setState({
-        ascending: !ascending,
-      })
-    } else {
-      this.setState({
-        sortBy: id,
-        ascending: true,
+      const sorted = [...rows].sort((a, b) => {
+        return a[sortBy] > b[sortBy] ? 1 : a[sortBy] < b[sortBy] ? -1 : 0
       })
-    }
-  }
 
-  handleColTextAlignChange(id, value) {
-    this.setState(state => ({
-      colTextAligns: {
-        ...state.colTextAligns,
-        [id]: value
+      return ascending ? sorted : sorted.reverse()
+    }, [sortBy, ascending, rows])
+
+    const handleSort = (event, { id }) => {
+      if (id === sortBy) {
+        setAscending(!ascending)
+      } else {
+        setSortBy(id)
+        setAscending(true)
       }
-    }))
-  }
+    }
 
-  renderHeaderRow(direction) {
-    const { headers } = this.props
-    const { colTextAligns , sortBy } = this.state
+    const handleColTextAlignChange = (id, value) => {
+      setColTextAligns((prevState) => ({
+        ...prevState,
+        [id]: value
+      }))
+    }
 
-    return (
+    const renderHeaderRow = (direction) => (
       <Table.Row>
-        {(headers || []).map(({ id, text, width }) => (
+        {headers.map(({ id, text, width }) => (
```

Comment:
Here there is truthy check i in the class-based example "((headers || []).map(...", which is missing from here.

Reply:
Done